docs/scripts/util/checkLink.ts

```diff
@@ -46,12 +49,12 @@ async function returnStatus({
      * e.g. 308 from "/guides" passes (because it's redirected to "react/guides"),
      *  but 308 from "connected-components/authenticator/headless" fails and returns a 404 ((because it's redirected to "react/connected-components/authenticator/advanced").
      */
-    if (statusCode === 308) {
-      const hostNameRegex = RegExp(`http(s)?:\/\/[^/]*`, 'i'); // matches everything between http(s)?: to "/", which is the hostname. e.g., "https://github.com/".
-      const platform = pageUrl.replace(hostNameRegex, '').split('/')[1];
-      const newHref = `${
-        href.match(hostNameRegex)[0]
-      }/${platform}${href.replace(hostNameRegex, '')}`;
+    if (statusCode === 308 && isInternalLink(href)) {
```

Comment:
In the comment on line 46, it says "check if it's an internal redirection...", but there was not logic for that, which caused the bug mentioned in the PR description.

So, add `isInternalLink` here